src/main/java/de/dhbw/karlsruhe/students/mailflow/external/infrastructure/preferences/FileUserRepository.java

```diff
@@ -84,13 +84,18 @@ private void loadUsers() throws LoadingUsersException, SaveUserException {
   @Override
   public Optional<User> findByEmail(Address email) throws LoadingUsersException, SaveUserException {
     loadUsers();
-    return users.stream().filter(user -> user.hasEmail(email)).findFirst();
+    return users.stream().filter(user -> user.hasAddress(email)).findFirst();
   }
 
   @Override
   public boolean save(User user) throws SaveUserException, LoadingUsersException {
     loadUsers();
+
+    // forcefully use the given object, e.g. if the password has changed even if equals() say
+    // they're equal (because of the same addresses used)
+    users.remove(user);
```

Comment:
I guess this does not throw anything, when there are no users right? It probably just returns false?

Reply:
yessss... see https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Set.html#remove(java.lang.Object)